httpcore/_sync/connection_pool.py

```diff
@@ -224,7 +224,7 @@ def request(
                 )
             except NewConnectionRequired:
                 connection = None
-            except Exception:  # noqa: PIE786
+            except BaseException:  # noqa: PIE786
```

Comment:
```suggestion
            except BaseException:  # noqa: PIE786
                # See https://github.com/encode/httpcore/pull/305 for motivation
                # behind catching 'BaseException' rather than 'Exception' here.
```